src/Playwright/Transport/WebSocketTransport.cs

```diff
@@ -0,0 +1,183 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Net.WebSockets;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Playwright.Transport
+{
+    internal class WebSocketTransport : IConnectionTransport, IDisposable
+    {
+        private const int DefaultBufferSize = 16000;  // Byte buffer size
+        private readonly ClientWebSocket _webSocket;
+        private readonly string _wsEndpoint;
+        private readonly CancellationTokenSource _readerCancellationSource;
+        private readonly CancellationTokenSource _connectCancellationSource;
```

Comment:
What's up with all these GH warnings?

Reply:
Ahh yes the object needs to be disposed.

Reply:
fixed